tendrl/gluster_integration/objects/volume/atoms/delete/__init__.py

```diff
@@ -88,8 +88,23 @@ def run(self):
             try:
                 # Acquire lock before deleting the volume from etcd
                 # We are blocking till we acquire the lock
+                # the lock will live for 60 sec after which it will be released.
                 lock = etcd.Lock(NS.etcd_orm.client, 'volume')
-                lock.acquire(blocking=True,lock_ttl=None)
+
+                while not lock.is_acquired:
+                    try:
+                        # with ttl set, lock will be blocked only for 60 sec
+                        # after which it will raise lock_expired exception.
+                        # if this is raised, we have to retry for lock
+                        lock.acquire(blocking=True,lock_ttl=60)
+                        if lock.is_acquired:
```

Comment:
The lock is refreshed during the acquire call itself, I dont think this would be a problem